modules/oe_whitelabel_extra_project/oe_whitelabel_extra_project.module

```diff
@@ -144,5 +160,147 @@ function _oe_whitelabel_extra_project_preprocess_inpage_nav(array &$variables):
       'label' => $node->{$item['#field_name']}->getFieldDefinition()->getLabel(),
     ];
   }
+}
+
+/**
+ * Adds variables for the project status.
+ *
+ * @param array $variables
+ *   Variables from hook_preprocess_node().
+ */
+function _oe_whitelabel_extra_project_preprocess_status_and_progress(array &$variables): void {
+  /** @var \Drupal\node\NodeInterface $node */
+  $node = $variables['node'];
+  /** @var \Drupal\datetime_range\Plugin\Field\FieldType\DateRangeItem|null $date_range_item */
+  $date_range_item = $node->get('oe_project_dates')->first();
+  if ($date_range_item === NULL || !$date_range_item->value || !$date_range_item->end_value) {
+    // One of the fields is empty.
+    return;
+  }
+
+  // Dates only store the date, not the time.
+  // Use the site-wide configured timezone, not a user-specific timezone.
```

Comment:
This is actually not aligned with the dates we are rendering below. Drupal doesn't take timezone into consideration when rendering date only fields.
This makes everything harder, since we have the dilemma of "is the date referring the day for the site timezone or my day" or "why this progress is not reflecting the other website dates which are using my user timezone"
or "why the event is over while day X is still in progress".
I'd say we remove this for now completely. Let's keep the utc dates, do something on the lines of
```
  $start = $date_range_item->start_date->setTime(0, 0, 0, 0);
  $end = $date_range_item->end_date->setTime(23, 59, 59, 0);
  $t_start = $start->getTimeStamp();
  $t_end = $end->getTimeStamp();
```
And that's it. We will follow up this to see how to tackle.

Reply:
I am very conflicted about this, also DGCOMM is planning to add timezones to this field so it can be made clear.

Reply:
The scenario is this:
- Site timezone is Brussels,BE.
- A logged-in user has a custom timezone in Sydney,Australia. Depending on site settings, the php timezone will be Sydney, not Brussels, when the request is processed.
- An anonymous user from Sydney has their own local time, but Drupal does not know about it. The php timezone will be Brussels, even though the anon user is in Sydney.

The events only store a date string without time.
We assume that the events happen in Brussels, so they begin and end at 00:00 in Brussels timezone.

With the current calculation, for both of the users from Sydney, the event would show up as "Planned" in the morning, and "Ongoing" in the evening.
If a user from California, one from Brussels, and one from Sydney, logged in or not, have a phone call while visiting the website, they will see the same project status.

If we use user timezone (which will happen if we use php timezone), then a user from California would see a different event status than a user from Sydney, and the anonymous visitor would see a different status than the logged-in user from Sydney.

> This is actually not aligned with the dates we are rendering below.
Indeed, the status would not conform with the date a user would see in their local clock.
However, this is already the case for anon visitors.

Reply:
tested again, I agree its good, the main thing that gives me confidence is that the javascript Date.now() is the same no matter the users location or system settings, so it will be the same calculation